framework/src/userobject/DomainUserObject.C

```diff
@@ -0,0 +1,113 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "DomainUserObject.h"
+#include "MooseVariableFE.h"
+#include "SubProblem.h"
+#include "Assembly.h"
+
+InputParameters
+DomainUserObject::validParams()
+{
+  InputParameters params = UserObject::validParams();
+  params += BlockRestrictable::validParams();
```

Comment:
so is this inheriting BoundaryRestrictable in the end?

Reply:
It should be still considered as block restricted. It operates on a subdomain, but includes evaluations one the sides of elements in the subdomain.